Build breaks on musl/alpine due to use of "stat64"

When building swan-station under Alpine Linux, this error will occur and the build will fail:

```
 1531 |   struct stat64 sysStatData;
      |                 ^~~~~~~~~~~
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1532:7: error: 'fstat64' was not declared in this scope; did you mean 'stat64'?
 1532 |   if (fstat64(fd, &sysStatData) < 0)
      |       ^~~~~~~
      |       stat64
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp: In function 'bool FileSystem::FileExists(const char*)':
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1573:17: error: aggregate 'FileSystem::FileExists(const char*)::stat64 sysStatData' has incomplete type and cannot be defined
 1573 |   struct stat64 sysStatData;
      |                 ^~~~~~~~~~~
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1574:32: error: invalid use of incomplete type 'struct FileSystem::FileExists(const char*)::stat64'
 1574 |   if (stat64(Path, &sysStatData) < 0)
      |                                ^
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1573:10: note: forward declaration of 'struct FileSystem::FileExists(const char*)::stat64'
 1573 |   struct stat64 sysStatData;
      |          ^~~~~~
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp: In function 'bool FileSystem::DirectoryExists(const char*)':
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1602:17: error: aggregate 'FileSystem::DirectoryExists(const char*)::stat64 sysStatData' has incomplete type and cannot be defined
 1602 |   struct stat64 sysStatData;
      |                 ^~~~~~~~~~~
/home/oliver/src/libretro-super/libretro-swanstation/src/common/file_system.cpp:1603:32: error: invalid use of incomplete type 'struct FileSystem::DirectoryExists(const char*)::stat64'
 1603 |   if (stat64(Path, &sysStatData) < 0)
      |                                ^
```

This is because Alpine Linux uses the musl implementation of libc, which does not have
"struct stat64".

I don't know much about the configuration system, But I'm assuming there's a "if linux" check that
allows it if and only if swanstation is being built on linux. (Windows I don't think has stat64)

--- a/src/common/memory_arena.cpp
+++ b/src/common/memory_arena.cpp
@@ -163,9 +163,9 @@ bool MemoryArena::Create(size_t size, bool writable, bool executable)
   shm_unlink(file_mapping_name.c_str());
 
   // ensure it's the correct size
-  if (ftruncate64(m_shmem_fd, static_cast<off64_t>(size)) < 0)
+  if (ftruncate(m_shmem_fd, static_cast<off_t>(size)) < 0)
   {
-    Log_ErrorPrintf("ftruncate64(%zu) failed: %d", size, errno);
+    Log_ErrorPrintf("ftruncate(%zu) failed: %d", size, errno);
     return false;
   }
 

